packages/kit/src/runtime/client/router.js

```diff
@@ -190,12 +190,12 @@ export class Router {
 			// Removing the hash does a full page navigation in the browser, so make sure a hash is present
 			const [base, hash] = url.href.split('#');
 			if (hash !== undefined && base === location.href.split('#')[0]) {
-				// Call `pushState` to add url to history so going back works.
-				// Also make a delay, otherwise the browser default behaviour would not kick in
-				setTimeout(() => history.pushState({}, '', url.href));
+				event.preventDefault();
 				const info = this.parse(url);
 				if (info) {
-					return this.renderer.update(info, [], false);
+					await this.renderer.update(info, [], false);
```

Comment:
do we actually need to call `renderer.update`? I think we just need to update the hash in the page store

Reply:
and if we decide to let renderer.update stay there, I think we should use `.then()` instead of `await` because it's the only need for `async` function. 
But as you said, I don't think it's actually needed, because we are re-rendering the same page, and that's not how browser normally behaves(when in not SPA environment) so we're not following the spec if we let it remain.

Reply:
@benmccann @Rich-Harris we can't do that because the stores don't react to that, we would have to trigger every store change manually, like I tried in https://github.com/sveltejs/kit/pull/3873/commits/6340fea3a8861de2ffc87ae168665456270d0ad0

Reply:
I would stick with `await` / `async` as I think it's easier to understand and more consistent with the rest of the code